### ISSUE TITLE
使用Npu训练完模型转换为静态图没有.pdmodel文件

### ISSUE THREAD

**Issue by OvO616** [PaddlePaddle/Paddle#72123]
### 请提出你的问题 Please ask your question

使用昇腾910B，paddlenlp训练完模型之后，使用paddle.jit.to_static以及paddle.jit.save函数，保存模型之后没有.pdmodel文件，只有.pdiparams文件和一个.json文件，我该如何使用paddle_infer.Config这个函数呢
芯片型号 昇腾 910B 架构是arm64架构 CANN版本 | CANN 8.0.RC1 使用如下安装的paddlecustom https://github.com/PaddlePaddle/PaddleCustomDevice/blob/develop/backends/npu/README_cn.md

**Comment by wangguan1995** [comment]
新版本去掉了pdmodel, 之前推理的代码【加载的部分】都不能用了

https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/jit/load_cn.html#load

**Comment by OvO616** [comment]
> 新版本去掉了pdmodel,

去掉了我如何使用这个函数呢，比如下面这段
config = paddle.inference.Config(model_file, params_file)  #之前需要传入模型文件跟.pdmodel文件，现在没有这个文件单独传入报错
 config.enable_custom_device("npu")

        # create predictor
self.predictor = paddle.inference.create_predictor(config)

**Comment by wangguan1995** [comment]
> > 新版本去掉了pdmodel,
> 
> 去掉了我如何使用这个函数呢，比如下面这段 config = paddle.inference.Config(model_file, params_file) #之前需要传入模型文件跟.pdmodel文件，现在没有这个文件单独传入报错 config.enable_custom_device("npu")
> 
> ```
>     # create predictor
> ```
> 
> self.predictor = paddle.inference.create_predictor(config)

直接paddle.jit.load, 不需要加载config了，都删了就行

**Comment by OvO616** [comment]
> > > 新版本去掉了pdmodel,
> > 
> > 
> > 去掉了我如何使用这个函数呢，比如下面这段 config = paddle.inference.Config(model_file, params_file) #之前需要传入模型文件跟.pdmodel文件，现在没有这个文件单独传入报错 config.enable_custom_device("npu")
> > ```
> >     # create predictor
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > self.predictor = paddle.inference.create_predictor(config)
> 
> 直接paddle.jit.load, 不需要加载config了，都删了就行

![Image](https://github.com/user-attachments/assets/4fffee48-defe-4d2f-9ed4-658007a2fdaa)好像还是报错啊 显示还是需要这个文件呢
我直接改的self.predictor = paddle.jit.load(params_file) 是这个形式把

**Comment by OvO616** [comment]
import os
import paddle
import numpy as np
import pandas as pd
import paddlenlp as ppnlp
from paddle import inference
from scipy.special import softmax
from paddlenlp.data import Tuple, Pad
import paddle.inference as paddle_infer

class Predictor(object):
    def __init__(self,
                 model_dir,
                 device="npu",
                 max_seq_length=256,
                 batch_size=32,
                 use_tensorrt=False,
                 precision="fp32",
                 cpu_threads=10,
                 enable_mkldnn=False):
        self.max_seq_length = max_seq_length
        self.batch_size = batch_size

        #model_file = model_dir + "/inference.pdmodel"
        params_file = model_dir + "/inference.pdiparams"
        #if not os.path.exists(model_file):
            #raise ValueError("not find model file path {}".format(model_file))
        if not os.path.exists(params_file):
            raise ValueError("not find params file path {}".format(params_file))
        config = paddle.jit.load(params_file)
        config.enable_custom_device("npu")

        # create predictor
        self.predictor = paddle.inference.create_predictor(config)
        self.input_handles = [
            self.predictor.get_input_handle(name)
            for name in self.predictor.get_input_names()
        ]
        self.output_handle = self.predictor.get_output_handle(
            self.predictor.get_output_names()[0])

        self.tokenizer = ppnlp.transformers.ErnieTokenizer.from_pretrained('ernie-3.0-medium-zh')

    def convert_example(self, example):
        '''
        文本特征提取
        '''
        text = example["text"]
        encoded_inputs = self.tokenizer(
            text=text, text_pair=None, max_seq_len=self.max_seq_length)
        input_ids = encoded_inputs["input_ids"]
        token_type_ids = encoded_inputs["token_type_ids"]
        return input_ids, token_type_ids

    def predict(self, data, batch_size=1, objective='binary', prob=0.8):
        '''
        对测试数据进行预测
        :param model: 模型实例
        :param data: 测试数据
        :param batch_size:批量大小
        :param objective:多分类（multiclass） or 二分类（binary）
        :param prob:二分类阈值大小
        :return:
        '''
        examples = []
        for text in data:
            example = {"text": str(text)}
            input_ids, token_type_ids = self.convert_example(example)
            examples.append((input_ids, token_type_ids))

        # Seperates data into some batches.
        batches = [
            examples[idx:idx + batch_size]
            for idx in range(0, len(examples), batch_size)
        ]
        batchify_fn = lambda samples, fn=Tuple(
            Pad(axis=0, pad_val=self.tokenizer.pad_token_id, dtype='int64'),  # input
            Pad(axis=0, pad_val=self.tokenizer.pad_token_type_id, dtype='int64'),  # segment
        ): fn(samples)

        results = []
        for batch in batches:
            input_ids, segment_ids = batchify_fn(batch)
            self.input_handles[0].copy_from_cpu(input_ids)
            self.input_handles[1].copy_from_cpu(segment_ids)
            self.predictor.run()
            logits = self.output_handle.copy_to_cpu()
            probs = softmax(logits, axis=1)
            idx = np.argmax(probs, axis=1)
            if objective == 'binary':
                idx = [1 if pro[1] > prob else 0 for pro in probs]
            elif objective == 'multiclass':
                idx = idx.tolist()
            results.extend(idx)
        return results


if __name__ == "__main__":
    # 获取相关参数
    paddle.set_device("npu")
    batch_size = 32

    # 读取数据
    df = pd.read_csv('/workdir/data/xw_5gxx/train/secondary_multiclass/dev.csv')
    data = df['text'].values.tolist()

    # 初始化模型
    predictor = Predictor(model_dir='/workdir/model/export/secondary_multiclass')

    batches = [
        data[idx:idx + batch_size]
        for idx in range(0, len(data), batch_size)
    ]

    # 开始预测
    y_pred = predictor.predict(data, batch_size=batch_size, objective='binary', prob=0.8)
    df['y_pred'] = pd.Series(y_pred)
    df.to_csv('result.csv', index=False)
或者您帮忙看看 现在是整个predictor部分都不需要了 还是怎么

**Comment by OvO616** [comment]
或者有什么版本是可以导出这个.pdmodel的吗 因为后续推理工作比如转onnx和om模型都需要这个

**Comment by LittleHeroZZZX** [comment]
设置环境变量禁用 PIR `export FLAGS_enable_pir_api=0`?

**Comment by 1357810la** [comment]
> import os import paddle import numpy as np import pandas as pd import paddlenlp as ppnlp from paddle import inference from scipy.special import softmax from paddlenlp.data import Tuple, Pad import paddle.inference as paddle_infer
> 
> class Predictor(object): def **init**(self, model_dir, device="npu", max_seq_length=256, batch_size=32, use_tensorrt=False, precision="fp32", cpu_threads=10, enable_mkldnn=False): self.max_seq_length = max_seq_length self.batch_size = batch_size
> 
> ```
>     #model_file = model_dir + "/inference.pdmodel"
>     params_file = model_dir + "/inference.pdiparams"
>     #if not os.path.exists(model_file):
>         #raise ValueError("not find model file path {}".format(model_file))
>     if not os.path.exists(params_file):
>         raise ValueError("not find params file path {}".format(params_file))
>     config = paddle.jit.load(params_file)
>     config.enable_custom_device("npu")
> 
>     # create predictor
>     self.predictor = paddle.inference.create_predictor(config)
>     self.input_handles = [
>         self.predictor.get_input_handle(name)
>         for name in self.predictor.get_input_names()
>     ]
>     self.output_handle = self.predictor.get_output_handle(
>         self.predictor.get_output_names()[0])
> 
>     self.tokenizer = ppnlp.transformers.ErnieTokenizer.from_pretrained('ernie-3.0-medium-zh')
> 
> def convert_example(self, example):
>     '''
>     文本特征提取
>     '''
>     text = example["text"]
>     encoded_inputs = self.tokenizer(
>         text=text, text_pair=None, max_seq_len=self.max_seq_length)
>     input_ids = encoded_inputs["input_ids"]
>     token_type_ids = encoded_inputs["token_type_ids"]
>     return input_ids, token_type_ids
> 
> def predict(self, data, batch_size=1, objective='binary', prob=0.8):
>     '''
>     对测试数据进行预测
>     :param model: 模型实例
>     :param data: 测试数据
>     :param batch_size:批量大小
>     :param objective:多分类（multiclass） or 二分类（binary）
>     :param prob:二分类阈值大小
>     :return:
>     '''
>     examples = []
>     for text in data:
>         example = {"text": str(text)}
>         input_ids, token_type_ids = self.convert_example(example)
>         examples.append((input_ids, token_type_ids))
> 
>     # Seperates data into some batches.
>     batches = [
>         examples[idx:idx + batch_size]
>         for idx in range(0, len(examples), batch_size)
>     ]
>     batchify_fn = lambda samples, fn=Tuple(
>         Pad(axis=0, pad_val=self.tokenizer.pad_token_id, dtype='int64'),  # input
>         Pad(axis=0, pad_val=self.tokenizer.pad_token_type_id, dtype='int64'),  # segment
>     ): fn(samples)
> 
>     results = []
>     for batch in batches:
>         input_ids, segment_ids = batchify_fn(batch)
>         self.input_handles[0].copy_from_cpu(input_ids)
>         self.input_handles[1].copy_from_cpu(segment_ids)
>         self.predictor.run()
>         logits = self.output_handle.copy_to_cpu()
>         probs = softmax(logits, axis=1)
>         idx = np.argmax(probs, axis=1)
>         if objective == 'binary':
>             idx = [1 if pro[1] > prob else 0 for pro in probs]
>         elif objective == 'multiclass':
>             idx = idx.tolist()
>         results.extend(idx)
>     return results
> ```
> 
> if **name** == "**main**": # 获取相关参数 paddle.set_device("npu") batch_size = 32
> 
> ```
> # 读取数据
> df = pd.read_csv('/workdir/data/xw_5gxx/train/secondary_multiclass/dev.csv')
> data = df['text'].values.tolist()
> 
> # 初始化模型
> predictor = Predictor(model_dir='/workdir/model/export/secondary_multiclass')
> 
> batches = [
>     data[idx:idx + batch_size]
>     for idx in range(0, len(data), batch_size)
> ]
> 
> # 开始预测
> y_pred = predictor.predict(data, batch_size=batch_size, objective='binary', prob=0.8)
> df['y_pred'] = pd.Series(y_pred)
> df.to_csv('result.csv', index=False)
> ```
> 
> 或者您帮忙看看 现在是整个predictor部分都不需要了 还是怎么
你好，请问这个问题解决了吗
